cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -368,6 +368,26 @@ List<ProgramHistory> getRuns(Collection<ProgramReference> programs, ProgramRunSt
    */
   int addApplication(ApplicationId id, ApplicationMeta meta) throws ConflictException;
 
+  /**
+   * Creates new application if it doesn't exist. Updates existing one otherwise.
+   *
+   * @param id            application id
+   * @param meta          application metadata to store
+   * @param markAsLatest  boolean, determines if the added application should be marked as latest
+   * @return              the number of edits to the application. A new application will return 0.
+   * @throws ConflictException if the app cannot be deployed when the user provided parent-version doesn't match the
+   *     current latest version
+   */
+  int addApplication(ApplicationId id, ApplicationMeta meta, boolean markAsLatest) throws ConflictException;
```

Comment:
would be better to add the flag to ApplicationMeta, otherwise we will end up with a method for every combination of options.

Reply:
Added the flag to ApplicationMeta.